keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/AdminService.kt

```diff
@@ -56,4 +63,38 @@ class AdminService(
       }
     }
   }
+
+  /**
+   * updates old artifact version records with their metadata, if exists, by type [deb/docker/npm]
+   */
+  fun backfillOldArtifactVersions(type: String) {
+    val artifactSupplier = artifactSuppliers.supporting(type)
+    log.debug("starting to backfill old artifacts versions with artifact metadata")
+    //1. get all registered artifacts
+    val deliveryArtifacts = repository.getAllArtifacts(type)
+    //2. for each artifact, fetch all versions
+    deliveryArtifacts.forEach { artifact ->
+      val versions = repository.artifactVersions(artifact)
+      versions.forEach { version ->
+        val status = repository.getReleaseStatus(artifact, version)
+        //don't update if metadata is already exists
+        if (repository.getArtifactBuildMetadata(artifact.name, type, version, status) == null) {
+          runBlocking {
```

Comment:
Consider running this `async` so that you'll do this in parallel instead of sequentially.